lib/image-stream/utils.js

```diff
@@ -16,47 +16,30 @@
 
 'use strict';
 
-const _ = require('lodash');
 const Bluebird = require('bluebird');
-const fs = Bluebird.promisifyAll(require('fs'));
-const archiveType = require('archive-type');
+const mime = require('mime-types');
 
 /**
  * @summary Get archive mime type
  * @function
  * @public
  *
- * @param {String} file - file path
+ * @param {String} filename - file path
  * @fulfil {String} - mime type
  * @returns {Promise}
  *
  * @example
- * utils.getArchiveMimeType('path/to/raspberrypi.img.gz').then((mimeType) => {
- *   console.log(mimeType);
- * });
+ * const mimeType = utils.getArchiveMimeType('path/to/raspberrypi.img.gz');
  */
-exports.getArchiveMimeType = (file) => {
-
-  // `archive-type` only needs the first 261 bytes
-  // See https://github.com/kevva/archive-type
-  const ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH = 261;
-
-  return Bluebird.using(fs.openAsync(file, 'r').disposer((fileDescriptor) => {
-    return fs.closeAsync(fileDescriptor);
-  }), (fileDescriptor) => {
-    const BUFFER_START = 0;
-    const chunk = new Buffer(ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH);
+exports.getArchiveMimeType = (filename) => {
+  // NOTE: For some awkward reason `mime-db` (used by `mime-types`)
+  // does not contain a file extension for `.gz`, so we need to test for it here
+  // TODO: Remove this once https://github.com/jshttp/mime-db/pull/68 is merged
+  if (/\.gz$/i.test(filename)) {
+    return 'application/gzip';
+  }
 
-    return fs.readAsync(
-      fileDescriptor,
-      chunk,
-      BUFFER_START,
-      ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH,
-      null
-    ).then(() => {
-      return _.get(archiveType(chunk), [ 'mime' ], 'application/octet-stream');
-    });
-  });
+  return mime.lookup(filename) || 'application/octet-stream';
```

Comment:
Yes, `mime-types` determines the mime type by file extension, which is very reliable as most of them are specced in RFCs by IANA / IETF / W3C / etc., and people usually don't just change file extensions around. It's also plenty faster than doing I/O to figure it out.

> Would there be any benefit into combining both approaches?

I don't think so, as `archive-type` is limited to certain file formats, and detection by magic number can be mighty unreliable when we get into the realms of other file formats (`.dmg`s for example are known to be notorious detection busters as they have no header and begin with the data-fork which can contain anything).
If we'd really want highest-grade format detection then we'll have to implement specific format tests (i.e. by having something like `handler.test( filename, callback )`) – but that's material for another time.